### DISCUSSION TITLE
Is it possible to import from a node_modules one level above?

### DISCUSSION THREAD

**Discussion by ghost** [preactjs/wmr#639] [Q&A] [answered]
During the development process, it can be convenient to keep the directory with common node_modules outside the project directory, a level or two higher. But the wmr cannot import files from it. Is this a bug or a feature?

**Comment by developit** [comment, marked as answer]
This is an oversight that we've been slowly working towards solving. In WMR 2, we moved the cache directory out of `node_modules` to allow for outside-of-CWD caching. Marvin also recently landed support for _aliasing_ things from outside/above the current working directory, but we don't yet do `node_modules` resolution that takes this into account. The last step is a fairly major rearchitecture of how we do streaming npm install, which includes using Node's built-in module resolution (which supports nested node_modules directories).

There are a few workarounds in the meantime:

1.  if this is a monorepo and you're using `yarn`, you can mark the packages you want to reference from wmr as `nohoist` in the root `package.json`:
    ```jsonc
    {
      "workspaces": {
        "packages": [ ... ],
        "nohoist": ["**/preact-iso", "**/other-package"]
      }
    }
    ```

2.  create a symlink to the parent node_modules directory:  `cd packages/wmr-based-site && ln -s ../../node_modules node_modules`
